### ISSUE TITLE
alt+arrow not working on bash on ubuntu on windows (wsl)

### ISSUE THREAD

**Issue by simonnier** [fish-shell/fish-shell#3910]
fish version 2.5.0

system: windows

terminal: bash on ubuntu on windows (wsl)

alt+right directly brings cursor to the end of line, and then alt+left or right moves by character not by word

While ctrl+arrow moves cursor by word in default mode. However, I want to use vi mode, so ctrl+arrow is not working in vi mode

**Comment by krader1961** [comment]
You can do this by adding the following to your `fish_user_key_bindings` function:

```
bind \e\[1\;5C forward-word
bind \e\[1\;5D backward-word
```

Whether that is something we should make a default is TBD. Also the question of whether that should be done in insert, normal, or both vi modes. Since Vim doesn't do anything sensible with those keys I'm disinclined to implement them as default bindings in fish.

**Comment by balabi0github** [comment]
@krader1961 sorry for late comment. But I tried your setting, it doesn't work in vi mode.

**Comment by faho** [comment]
@balabi0github: First of all, anything bound without an explicit mode should be bound in "default" mode, which in vi-talk would be called "normal".

Secondly, you'll have to check if your terminal actually sends that sequence. Execute `fish_key_reader` and press that combination. That'll show you how you'd have to bind it.

**Comment by krader1961** [comment]
I updated fish on Windows 10 WSL and ran `fish_key_reader` to confirm that [ctrl]+arrow-key does in fact send the sequences I documented in my earlier comment. And neither sequence is bound by default in the vi bindings but are in the default/emacs bindings (to `backward-word` and `forward-word`). It seems reasonable to add those to the vi normal mode bindings. Obviously they don't make sense for vi insert mode.
